blogposts/2020/performance-improvements-in-precise-code-intel.md

```diff
@@ -0,0 +1,252 @@
+---
+title: "Performance improvements in precise code intel"
+author: Eric Fritz
+authorUrl: https://eric-fritz.com
+publishDate: 2020-06-17T10:00-07:00
+tags: [blog]
+slug: performance-improvements-in-precise-code-intel
+published: true
+---
+
+In [Sourcegraph 3.16](https://about.sourcegraph.com/blog/sourcegraph-3.16#performance-improvements-for-precise-code-intelligence), we mentioned a rewrite of the services composing the precise-code-intel backend from TypeScript to Go. There were many reasons for this rewrite, but there's one particular reason that I'd like to explore here: _I know how to improve Go code - I'm not sure how to do the same for TypeScript_.
+
+This is **not** to say anything negative about TypeScript -- the language is a surprising joy to work in, I'll admit after nearly a year. This is **not** to say that it's impossible to write code that performs well in a Node.js environment -- V8 is a true beast of engineering. This is **not** to say that there is a lack of tools in order to help developers profile and improve their code.
+
+This **is** to say that I think it was good move to perform the rewrite to allow myself to play to my strengths. Rewriting this code in a language for which I have a better mental model of semantics and performance, a better grasp of the ecosystem, and years of _actual_ experience writing high-performance code allows me to move with enough velocity in the future that the time spent rewriting will be paid off in short order. As an added benefit, the rewrite also opened the code up to all other backend developers at Sourcegraph where Go is a core competency.
```

Comment:
I think it would be better to reframe this around our teams strengths rather than your own individual strengths. (not the other way around). We have a lot of engineers to know how to write and scale backend Go systems, but not the same for TS.

Reply:
Updating.